### DISCUSSION TITLE
Prisma Accelerate not working with Cloudfare and next.js

### DISCUSSION THREAD

**Discussion by CardCounter21** [prisma/prisma#23617] [Q&A] [answered]
### Question

I am creating a next.js website with prisma accelerate. Currently I am using Clerk for user authentication and connection to a planetscale backend. Whenever I try to run my app on local host I get this error message: 
<img width="949" alt="Screenshot 2024-03-25 at 3 39 29 PM" src="https://github.com/prisma/prisma/assets/160655497/963d402b-d31d-437a-906c-3f0f8ccf9422">

I have installed Prisma Accelerate exactly as the docs required. 
Steps I followed: 
1. I generated the api key and added it to my .env file. 
2. npm install @prisma/client@latest @prisma/extension-accelerate
3. Because I want to host my website on cloudfare I use: import { PrismaClient } from '@prisma/client/edge'

I have a db.ts: 
```ts
import { PrismaClient } from "@prisma/client/edge";
import { withAccelerate } from '@prisma/extension-accelerate'

const prisma = new PrismaClient().$extends(withAccelerate())

declare global {
    var prisma: PrismaClient | undefined;
}

export const db = globalThis.prisma || new PrismaClient()

if(process.env.NODE_ENV !== "production") globalThis.prisma = db
```
A initial-profile.ts: 
```ts
import { currentUser, redirectToSignIn } from "@clerk/nextjs";
import { db } from './db';

export const initialProfile = async () => {
    const user = await currentUser();

    if(!user){
        return redirectToSignIn;
    }

    const profile = await db.user.findUnique({
        where:{
            userId: user.id
        }
    });

    if(profile){
        return profile;
    }
    console.log('WORK')

    const newProfile = await db.user.create({
        data: {
            userId: user.id,
            name: `${user.firstName} ${user.lastName}`,
            imageUrl: user.imageUrl,
            email: user.emailAddresses[0].emailAddress,
            username: `${user.username}`,
            isAdmin: false,
        }
    })
    console.log('user created')

    return newProfile;
}
```
a schema.prisma: 
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "mysql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
}

model User {
  id        String   @id @default(uuid())
  userId    String   @unique
  name      String
  username  String
  imageUrl  String   @db.Text
  email     String   @db.Text
  isAdmin   Boolean
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```
If someone could give me some pointers I would be extremely grateful! 
Have a great day!

### Information about Prisma Schema, Client Queries and Environment (optional)

Next.js - setup for cloudfare deployment. However I am hosting on vercel currently for production, prisma accelerate, 

Here is what my file layout looks like currently: 
<img width="244" alt="Screenshot 2024-03-25 at 3 47 36 PM" src="https://github.com/prisma/prisma/assets/160655497/4f7d3f76-9b3b-4b26-bd17-bd7407ee28f4">

Schema.Primsa: 
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "mysql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
}

model User {
  id        String   @id @default(uuid())
  userId    String   @unique
  name      String
  username  String
  imageUrl  String   @db.Text
  email     String   @db.Text
  isAdmin   Boolean
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Widget {
  id        String @id @default(uuid())
  type      String @default("Lessons")
  profileId String

  @@index([profileId])
}

model Lessons {
  id          String   @id @default(uuid())
  isFree      Boolean
  name        String
  description String
  videoUrl    String
  author      String
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
}

model Groups {
  id        String   @id @default(uuid())
  name      String
  imageUrl  String   @db.Text
  profileId String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@index([profileId])
}

model Member {
  id        String     @id @default(uuid())
  role      MemberRole @default(NORMAL)
  profileId String
  serverId  String
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt

  @@index([profileId])
  @@index([serverId])
}

model Message {
  id        String      @id @default(uuid())
  content   String
  fileUrl   String?
  type      MessageType @default(TEXT)
  memberId  String
  serverId  String
  createdAt DateTime    @default(now())
  updatedAt DateTime    @updatedAt

  @@index([memberId])
  @@index([serverId])
}

enum MemberRole {
  ADMIN
  MODERATOR
  NORMAL
}

enum MessageType {
  TEXT
  IMAGE
  GAME
}
```

**Comment by nurul3101** [comment]
Hey @CardCounter21 👋 

Is your database publicly accessible?

Are you invoking the `npx prisma generate --no-engine` flag before running your code locally?

**Reply by CardCounter21** [reply]
Hey, I really appreciate your response, yes I did run that. I think It is an issue with prisma rate limiting my project as It does work when only a few people are working, however when there are many people working on the project the error happens. I will upgrade to the pro plan and see If the issue persists. Thank you!

**Reply by nurul3101** [reply, marked as answer]
If I understand correctly, then it is an intermittent issue. It doesn't happen all the time?
Did you exhaust the 60,000 free queries per month quota ([pricing](https://www.prisma.io/pricing))?

Sure, let me know in case the issue persists!

**Reply by CardCounter21** [reply]
@nurul3101 I found out what the error is. The school wifi detected that I was making too make query requests from my local host so they IP blocked me 🤣. It works perfectly fine when I use a vpn. Thank you so much for your attention and help tho!
